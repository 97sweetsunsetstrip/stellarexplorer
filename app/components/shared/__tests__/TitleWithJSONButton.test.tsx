import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import { TitleWithJSONButton } from '../TitleWithJSONButton'

describe('TitleWithJSONButton', () => {
  it('renders button with given url', () => {
    const url = 'https://somebackend.xyz/ledger/12345'
    const title = 'Ledger'
    render(<TitleWithJSONButton title={title} url={url} />)
    expect(screen.getByText('Ledger')).toBeInTheDocument()
    expect(screen.getByText('JSON')).toBeInTheDocument()

    // snapshot test
    const tree = renderer
      .create(<TitleWithJSONButton title={title} url={url} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
