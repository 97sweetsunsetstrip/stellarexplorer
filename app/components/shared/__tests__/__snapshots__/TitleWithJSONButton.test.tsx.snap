// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleWithJSONButton renders button with given url 1`] = `
<div>
  <span>
    Ledger
  </span>
  <span
    className="pull-right"
  >
    <span>
      <a
        className="backend-resource-badge-button"
        onClick={[Function]}
      >
        JSON
      </a>
    </span>
  </span>
</div>
`;
